### PR TITLE
Tx build/data aggregate

### PR DESCRIPTION
**Pull request by jordanlesich**
## GitHub Issue

https://github.com/HausDAO/daohaus-monorepo/issues/417

## Changes

Big list in issue

## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs locally
- [x] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

File: apps/summon-app/src/utils/summonlegos.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { LOCAL_ABI } from '@daohaus/abi-utilities';
+import { ContractLego, CONTRACTS } from '@daohaus/common-utilities';
+
+const BaalSummonerContract: ContractLego = {
+  contractName: 'BaalSummoner',
+  type: 'local',
+  abi: LOCAL_ABI.BAAL_SUMMONER,
+  keychain: CONTRACTS.V3_FACTORY,
```
Comment (skuhlmann) [inline apps/summon-app/src/utils/summonlegos.ts:8]: nice just having the keychain here

File: libs/common-utilities/src/types/general.ts [hunk 2]
```diff
@@ -2,3 +2,10 @@ export type Noun = {
   singular: string;
   plural: string;
 };
+
+// By definition this needs to use any.
+// But we should be dilligent about using this type and make sure that
+// in cases where we're using this to handle highly level functionality
+// we can extends with more specific types (ex. TXBuilder)
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type ArbitraryState = Record<string, any>;
```
Comment (skuhlmann) [inline libs/common-utilities/src/types/general.ts:11]: :+1:

File: libs/tx-builder-feature/src/utils/txBuilderUtils.ts [hunk 3]
```diff
@@ -199,6 +212,7 @@ export async function handleFireTx({
     provider.getSigner().connectUnchecked()
   );
 
-  const ethersTx = await contract.functions[txName](...args);
-  return executeTx(tx, ethersTx, setTransactions, chainId);
+  const ethersTx = await contract.functions[method](...realArgs);
```
Comment (skuhlmann) [inline libs/tx-builder-feature/src/utils/txBuilderUtils.ts:215]: Not needed right now, But I think this is likely were we want to send the gasLimit when we get to processProposal down the line.

Can use the overrides object:
https://docs.ethers.io/v5/api/contract/contract/#contract-functionsSend

**Review by skuhlmann — APPROVED** [review]

**Comment by skuhlmann** [comment]
Looking good.

File: libs/tx-builder-feature/src/utils/txBuilderUtils.ts (hunk 3 above)
Comment (jordanlesich) [reply libs/tx-builder-feature/src/utils/txBuilderUtils.ts:215]: Thinking the same. Am I correct in assuming that this is only crucial for when we call processProposal?

Comment (skuhlmann) [reply libs/tx-builder-feature/src/utils/txBuilderUtils.ts:215]: > Thinking the same. Am I correct in assuming that this is only crucial for when we call processProposal?

Yeah that is the only one we'll need it on.

**Pull request merged**